common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.players;
+
+import com.wynntils.core.components.Model;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.models.character.CharacterModel;
+import com.wynntils.models.players.type.GuildRank;
+import com.wynntils.utils.mc.LoreUtils;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+
+public class GuildModel extends Model {
+    private static final Pattern GUILD_NAME_MATCHER = Pattern.compile("§3(.*?)§b.*");
+    private static final Pattern GUILD_RANK_MATCHER = Pattern.compile("§7Rank: §f(.*)");
+
+    private String name = "";
+    private GuildRank rank;
+
+    public GuildModel(CharacterModel characterModel) {
+        super(List.of(characterModel));
+    }
+
+    public void parseGuildInfoFromGuildMenu(ItemStack guildInfoItem) {
+        List<StyledText> lore = LoreUtils.getLore(guildInfoItem);
+
+        for (StyledText line : lore) {
+            Matcher guildNameMatcher = line.getMatcher(GUILD_NAME_MATCHER);
+            if (guildNameMatcher.matches()) {
+                name = guildNameMatcher.group(1);
+                continue;
```

Comment:
Is name always before rank? Otherwise we will fail to get a rank. Perhaps move the rank block above the name block so they get in the "logical" order, and add a comment stating that this is so.

Reply:
I think so, unless someone is able to check if they appear differently for different ranks

Reply:
Yeah name before rank
![image](https://github.com/Wynntils/Artemis/assets/8337467/b5139b90-2517-408e-87ef-7b4fbf0ef20e)